crates/miden-lib/src/account/components/mod.rs

```diff
@@ -40,3 +46,72 @@ pub fn rpo_falcon_512_library() -> Library {
 pub fn basic_fungible_faucet_library() -> Library {
     BASIC_FUNGIBLE_FAUCET_LIBRARY.clone()
 }
+
+// WELL KNOWN COMPONENTS
+// ================================================================================================
+
+/// The enum holding the types of basic well-known account components provided by the `miden-lib`.
+pub enum WellKnownComponents {
+    BasicWallet,
+    BasicFungibleFaucet,
+    RpoFalcon512,
+}
+
+impl WellKnownComponents {
+    /// Returns the procedure digests vector, containing digests of all procedures provided by the
+    /// current component.
+    fn procedure_digests(&self) -> Vec<Digest> {
+        match self {
+            Self::BasicWallet => basic_wallet_library().mast_forest().procedure_digests().collect(),
+            Self::BasicFungibleFaucet => {
+                basic_fungible_faucet_library().mast_forest().procedure_digests().collect()
+            },
+            Self::RpoFalcon512 => {
+                rpo_falcon_512_library().mast_forest().procedure_digests().collect()
+            },
+        }
+    }
```

Comment:
Could we not write this as?:
```rust
fn procedure_digests(&self) -> impl Iterator<Item = Digest> {
    let forest = match self {
        Self::BasicWallet => BASIC_WALLET_LIBRARY.mast_forest(),
        Self::BasicFungibleFaucet => BASIC_FUNGIBLE_FAUCET_LIBRARY.mast_forest(),
        Self::RpoFalcon512 => BASIC_FUNGIBLE_FAUCET_LIBRARY.mast_forest(),
    };

    forest.procedure_digests()
}
```
This way, we won't need to allocate a vector when iterating over procedures.